lib/yaml_vault/yaml_tree_builder.rb

```diff
@@ -74,7 +76,9 @@ def scalar(value, anchor, tag, plain, quoted, style)
           else
             result.value = @cryptor.encrypt(value)
           end
+          result.value = handle_prefix(result.value)
```

Comment:
the method name seems to be different from the actual process.
I suggest `add_prefix_and_suffix` and `remove_prefix_and_suffix`.